src/main/java/jenkins/scm/api/SCMEvent.java

```diff
@@ -198,10 +208,13 @@ protected SCMEvent(SCMEvent<P> copy) {
      * @return a {@link ScheduledExecutorService}.
      */
     @NonNull
-    protected static ScheduledExecutorService executorService() {
-        // Future-proofing, if we find out that events are drowning Timer then we may need to move them to their
-        // own dedicated ScheduledExecutorService thread pool
-        return Timer.get();
+    protected static synchronized ScheduledExecutorService executorService() {
+        if (executorService == null) {
+            // corePoolSize is set to 10, but will only be created if needed.
+            // ScheduledThreadPoolExecutor "acts as a fixed-sized pool using corePoolSize threads"
+            executorService = new ImpersonatingScheduledExecutorService(new ScheduledThreadPoolExecutor(10, new NamingThreadFactory(new ClassLoaderSanityThreadFactory(new DaemonThreadFactory()), "SCMEvent")), ACL.SYSTEM);
+        }
+        return executorService;
     }
 
```

Comment:
we'll need a `@Terminator` annotated static method that shuts down this pool to play nicely with servlet containers that get reused, e.g. if anyone is in a downstream plugin and has lots of event unit tests, they may end up with *lots* of these thread pools. Better to expressly shut them down with the Jenkins lifecycle. 

See https://github.com/jenkinsci/jenkins/blob/master/core/src/main/java/hudson/init/Terminator.java